packages/shorebird_cli/lib/src/shorebird_flutter.dart

```diff
@@ -133,33 +133,40 @@ class ShorebirdFlutter {
   /// Converts a full git revision to a short revision string.
   String shortRevisionString(String revision) => revision.substring(0, 10);
 
+  /// Given a revision and a version, formats them into a single string.
+  ///
+  /// e.g. 3.16.3 and b9b2390296b9b2390296 -> 3.16.3 (b9b2390296)
+  String formatVersion({required String revision, required String? version}) {
+    version ??= 'unknown';
+    return '$version (${shortRevisionString(revision)})';
+  }
+
   /// Returns the current Shorebird Flutter version and revision.
   /// Returns unknown if the version check fails.
   Future<String> getVersionAndRevision() async {
-    String? version = 'unknown';
+    late final String? version;
+
     try {
       version = await getVersionString();
-    } catch (_) {}
+    } on ProcessException {
+      version = 'unknown';
+    }
 
-    return '$version (${shortRevisionString(shorebirdEnv.flutterRevision)})';
+    return formatVersion(
+      version: version,
+      revision: shorebirdEnv.flutterRevision,
+    );
   }
 
   /// Returns the current Shorebird Flutter version.
   /// Throws a [ProcessException] if the version check fails.
   /// Returns `null` if the version check succeeds but the version cannot be
   /// parsed.
-  Future<String?> getVersionString({String? revision}) async {
-    final result = await git.forEachRef(
-      contains: revision ?? shorebirdEnv.flutterRevision,
-      format: '%(refname:short)',
-      pattern: 'refs/remotes/origin/flutter_release/*',
-      directory: _workingDirectory(),
+  Future<String?> getVersionString() async {
```

Comment:
Maybe `getVersionForCurrentRevision` for clarity?

Reply:
Sounds good! Do should we rename `getVersionAndRevision` to `getCurrentVersionAndRevision` (or something in those lines) too? So both methods have similar naming?

Reply:
Sounds good!